tensorflow/core/kernels/mkl/onednn_fused_matmul_ops_test.cc

```diff
@@ -0,0 +1,749 @@
+      inputs.push_back({"output_min", 0, DT_FLOAT});
+      inputs.push_back({"output_max", 0, DT_FLOAT});
+      extended_fused_ops.push_back("Requantize");
+      output_dtypes = {out_dtype, DT_FLOAT, DT_FLOAT};
+    } else {
+      extended_fused_ops.push_back("Dequantize");
+      output_dtypes = {out_dtype};
+    }
+
+    TF_EXPECT_OK(NodeDefBuilder("quantized_fused_matmul", "_QuantizedMatMul")
+                     .Attr("Tdevice_inputs", std::vector<DataType>())
+                     .Input(FakeInput())
+                     .Input(inputs)
+                     .Attr("Thost_outputs", output_dtypes)
+                     .Attr("Tdevice_outputs", std::vector<DataType>())
+                     .Attr("T1", qinput_dtype)
+                     .Attr("T2", DT_QINT8)
+                     .Attr("Tbias", real_dtype)
+                     .Attr("Tout", out_dtype)
+                     .Attr("U", real_dtype)
+                     .Attr("transpose_a", transpose_x)
+                     .Attr("transpose_b", transpose_y)
+                     .Attr("fused_ops", extended_fused_ops)
+                     .Attr("leakyrelu_alpha",
+                           has_leaky_relu ? this->leakyrelu_alpha_ : 0.2f)
+                     .Attr("input_quant_mode", input_quant_mode)
+                     .Attr("output_quant_mode", output_quant_mode)
+                     .Finalize(&fused_matmul));
+    if (requantize) {
+      NodeDef dequantize;
+      TF_EXPECT_OK(NodeDefBuilder("dequantize", "Dequantize")
+                       .Input({"quantized_fused_matmul", 0, out_dtype})
+                       .Input({"quantized_fused_matmul", 1, DT_FLOAT})
+                       .Input({"quantized_fused_matmul", 2, DT_FLOAT})
+                       .Attr("dtype", real_dtype)
+                       .Attr("mode", output_quant_mode)
+                       .Finalize(&dequantize));
+      add_nodes = {&fused_matmul, &dequantize};
+      RunAndFetch(root, {dequantize.name()}, &outputs, add_nodes);
+    } else {
+      add_nodes = {&fused_matmul};
+      RunAndFetch(root, {fused_matmul.name()}, &outputs, add_nodes);
+    }
+    *result = outputs[0];
+  }
+
+  template <typename FusedGraphRunner>
+  void VerifyTensorsNear(const std::vector<int64_t>& x_dims,
+                         const std::vector<int64_t>& y_dims,
+                         const FusedOpsAndDims& fused_ops_and_dims,
+                         const GraphRunner& run_default,
+                         const FusedGraphRunner& run_fused, bool transpose_x,
+                         bool transpose_y, const double atol = 1e-5,
+                         // The following arguments are used by quantized fusion
+                         string input_quant_mode = "SCALED",
+                         string output_quant_mode = "SCALED",
+                         bool is_bias_quantized = false,
+                         bool is_perchannel = false, bool requantize = false) {
+    srand(1234);
```

Comment:
Please don't rely on this.  Our random number generation sometimes changes, so this won't guarantee test stability, and likely leads to incorrect tolerances.

Reply:
Removed `srand`. Thanks!